### PR TITLE
update lights function

### PR DESCRIPTION
**Pull request by vulongphan**
Resolves #5027

 Changes:

Update `light.js` file

 Screenshots of the change:

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [ ] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

**Comment by vulongphan** [comment]
@limzykenneth @stalgiag I have checked the `prettier` version in the `package.json` file and it is the same version as the one in `main`. I have also created this PR which is directly branched from `main` but the issue still persists. Do you have any suggestions?

**Comment by limzykenneth** [comment]
@vulongphan I'm thinking more in terms of the prettier version that is used by other tools on your system. The VSCode plugin for example doesn't work with our version of prettier so maybe something like that packages it's own prettier or using prettier that is installed globally, etc.

**Comment by vulongphan** [comment]
@limzykenneth I really do agree with your speculation. It is true that the prettier version of VSCode I am using does not work with our prettier version. Hence, I have made some appropriate changes in formating and now lint has passed.

Thank you!

File: src/webgl/light.js [hunk 1]
```diff
@@ -409,8 +409,20 @@ p5.prototype.pointLight = function(v1, v2, v3, x, y, z) {
  */
 p5.prototype.lights = function() {
   this._assert3d('lights');
-  this.ambientLight(128, 128, 128);
-  this.directionalLight(128, 128, 128, 0, 0, -1);
+  if (this._colorMode !== 'rgb') {
```
Comment (lmccart) [inline src/webgl/light.js:412]: could you update this to `constants.RGB`? here's an example of how it's done by importing constants and then calling it in [this file](https://github.com/processing/p5.js/blob/bb381febc246c7599381801ae314b3d40330d9d8/src/webgl/3d_primitives.js#L1078)

Comment (lmccart) [reply src/webgl/light.js:412]: same thing on line 416

**Comment by stalgiag** [comment]
Thanks for working through this @vulongphan . I see that you run into the same problem as #4998 . Really impressed that you powered through.

@lmccart 's point about the color mode constants reminded me that we only have two non-RGB color modes to think about and they both work similarly when making white, so we can actually avoid the whole array copying stuff by just doing something akin to this:

```js
  this._assert3d('lights');
  if (this._colorMode === constants.RGB) {
    this.ambientLight(128, 128, 128);
    this.directionalLight(128, 128, 128, 0, 0, -1);
  } else {
    const maxBright = this._colorMaxes[this._colorMode][2];
    this.ambientLight(0, 0, maxBright);
    this.directionalLight(0, 0, maxBright, 0, 0, -1);
  }
```

Thanks again for sticking around for this entire saga!

**Comment by vulongphan** [comment]
@stalgiag @lmccart I have incorporated both of your feedback. Lint has passed as well. Please check the most recent push [5c0cdaa](https://github.com/processing/p5.js/pull/5049/commits/5c0cdaa362998d03a620eaf44b52b8b3335e0243)

Thank you!

**Comment by stalgiag** [comment]
Amazing! Thank you for your perserverance @vulongphan !

@all-contributors please add @vulongphan for code

**Pull request merged**